### PR TITLE
#169817864 Validate signup data

### PR DESCRIPTION
**Pull request by VedaDom**
#### What does this PR do?
* This adds validation to signup endpoint.
* This checks if email provided by new use is already taken.
#### Description of Task to be completed?
This requires user to provide the following fields:
`{
      firstname: '',
      lastname: '',
      email: '',
      password: '',
      country: '',
      gender: '',
      birthday: ''
  }`
#### How should this be manually tested?
* Clone it
* Run npm i to install all dependencies
* Run npm start
* Browse this local url `localhost:3000`
#### What are the relevant pivotal tracker stories?
- [Pivotal tracker story #169817864](https://www.pivotaltracker.com/story/show/169817864)

### CODE REVIEW

File: src/database/models/index.js [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
-  .readdirSync(__dirname)
-  .filter(file => {
-    return (file.indexOf('.') !== 0) && (file !== basename) && (file.slice(-3) === '.js');
-  })
-  .forEach(file => {
-    const model = sequelize['import'](path.join(__dirname, file));
+const sequelize = new Sequelize(config.use_env_variable, config);
+
+// if (config.use_env_variable) {
+//   sequelize = new Sequelize(process.env[config.use_env_variable], config);
```
Comment (aanchirinah) [inline src/database/models/index.js]: ### Nitpick

- Remove commented code when it's not gonna be used or put a TODO comment on it.

File: src/helpers/validate.helper.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { check } from 'express-validator';
+
+/**
+ * A class that contains functions for all validations
+ */
+class Validate {
+  /**
+ * Adds two numbers together.
```
Comment (aanchirinah) [inline src/helpers/validate.helper.js]: ### Critical

- Is that an appropriate description?

File: src/middlewares/auth.middleware.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import UserServices from '../services/user.service';
+
+const isEmailUsed = async (req, res, next) => {
+  const isEmailExist = await UserServices.isEmailExist(req.body.email);
```
Comment (aanchirinah) [inline src/middlewares/auth.middleware.js]: ### Nitpick

- Include some sort of exception handling.

File: src/services/user.service.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import db from '../database/models';
+/**
+ * A class that contains functions for all user services
+ */
+class UserServices {
+/**
+ * Adds two numbers together.
```
Comment (aanchirinah) [inline src/services/user.service.js]: ### Critical

- Not an appropriate description.

File: src/tests/index.js [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
 
 describe('App Tests', () => {
-   it('should return the homepage', (done) => {
-      chai.request(app).get('/')
-      .end((err,res) => {
-         res.should.have.status(200);
-         done();
-      })
-   });
+  it('should return the homepage', (done) => {
```
Comment (aanchirinah) [inline src/tests/index.js:9]: ### Nitpick

- Maybe more tests for what you worked on.

**Review by aanchirinah — CHANGES_REQUESTED** [review]
Good job, kindly attend to my changes.

File: src/helpers/validate.helper.js (hunk 2 above)
Comment (VedaDom) [reply src/helpers/validate.helper.js]: I apologise for this mistake.  I forgot to edit this comment I got from jsDoc documentation website. it wont happen again.

File: src/tests/index.js (hunk 5 above)
Comment (VedaDom) [reply src/tests/index.js:9]: My tests are now in `signup.test.js` but are commented due to they can't pass without signup route.
I have tested them locally and they are all passing.

**Pull request merged**